Sources/StreamChat/StateLayer/CurrentUser.swift

```diff
@@ -0,0 +1,105 @@
+    
+    /// Updates the currently logged-in user data.
+    ///
+    /// - Note: Setting any arguments to nil will keep the existing value.
+    ///
+    /// - Parameters:
+    ///   - name: The name to be set to the user.
+    ///   - imageURL: The URL of the avatar image.
+    ///   - extraData: Additional data associated with the user.
+    ///
+    /// - Throws: An error while communicating with the Stream API or when user is not logged in.
+    public func updateUserData(name: String? = nil, imageURL: URL? = nil, extraData: [String: RawJSON] = [:]) async throws {
+        try await currentUserUpdater.updateUserData(currentUserId: try currentUserId(), name: name, imageURL: imageURL, userExtraData: extraData)
+    }
+    
+    // MARK: - Managing Channels
+    
+    /// Mark all the user's channels as read.
+    ///
+    /// - Throws: An error while communicating with the Stream API or when user is not logged in.
+    public func markAllChannelsRead() async throws {
+        try await currentUserUpdater.markAllRead(currentUserId: try currentUserId())
+    }
+    
+    // MARK: - Managing User Devices
+    
+    /// Loads an array of devices associated with the logged-in user.
+    ///
+    /// - Note: Devices can be read later from ``CurrentUserState.devices``.
+    ///
+    /// - Throws: An error while communicating with the Stream API or when user is not logged in.
+    /// - Returns: An array of devices receiving push notifications.
+    public func loadDevices() async throws -> [Device] {
+        try await currentUserUpdater.fetchDevices(currentUserId: try currentUserId())
+    }
+    
+    /// Registers the current user's device for push notifications.
+    ///
+    /// Registering a device associates it with the user and tells the push provider to send new message notifications to that device.
+    ///
+    /// - Tip: Register the user's device for remote push notifications once your user is successfully connected to Chat.
+    ///
+    /// - Parameter device: The device information required for registering the device. Use ``PushDevice.apn(token:providerName:)`` for APN.
+    ///
+    /// - Throws: An error while communicating with the Stream API or when user is not logged in.
+    public func addDevice(_ device: PushDevice) async throws {
+        try await currentUserUpdater.addDevice(device, currentUserId: try currentUserId())
+    }
+    
+    /// Removes the specified device from the current user.
+    ///
+    /// Unregistering a device removes the device from the user and stops further new message notifications.
+    ///
+    /// - Parameter deviceId: The id of the device to unregister from push notifications.
+    ///
+    /// - Throws: An error while communicating with the Stream API or when user is not logged in.
+    public func removeDevice(_ deviceId: DeviceId) async throws {
+        try await currentUserUpdater.removeDevice(id: deviceId, currentUserId: try currentUserId())
+    }
+    
+    // MARK: -
```

Comment:
Typo?
```suggestion
    // MARK: - Helpers
```

Reply:
Actually it wasn't, just used it for creating a line without text. It was kind of `// MARK: - Private` or helper like you suggested. Does not hurt adding a text to the separator as well.